pkg/util/template/funcerror.go

```diff
@@ -93,17 +93,16 @@ func getError(apiErr *apierrors.APIError, getErrInput *GetErrorInput) *apierrors
 	if apiErr == nil {
 		return nil
 	}
+	apiErrByReason := getErrorByReason(apiErr, getErrInput.ByReason)
+	if apiErrByReason != nil {
+		return apiErrByReason
+	}
```

Comment:
I thought by_location and by_reason are independent. But in your implementation they are not.

Reply:
> I thought by_location and by_reason are independent. 

Should they be independent? 

What if the caller wants to match `ValidationFailed` without providing any location, as seen in `setup_oob_otp.html`? 🙏

Reply:
ok, thanks daic!